cpp/oneapi/dal/backend/interop/table_conversion.hpp

```diff
@@ -39,6 +40,7 @@ struct daal_array_owner {
 
 template <typename T>
 inline auto allocate_daal_homogen_table(std::int64_t row_count, std::int64_t column_count) {
+    detail::check_mul_overflow(row_count, column_count);
```

Comment:
```suggestion
    dal::detail::check_mul_overflow(row_count, column_count);
```

Recommend to use fully qualified name for all common functionality in `detail`, because we could have it's own `detail` in the namespace nested to `dal`. For example, `dal::pca::detail`, so `detail` would mean `pca::detail`. 

Technically, in this case, `dal::detail` is not required, but I want to have consistency in `dal::detail` usage across the library.